internal/services/elastic/elastic_tag_rule_resource.go

```diff
@@ -0,0 +1,308 @@
+package elastic
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/hashicorp/terraform-provider-azurerm/helpers/azure"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/elastic/legacysdk/elastic/mgmt/2020-07-01/elastic"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/elastic/parse"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/elastic/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/timeouts"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+func resourceElasticTagRule() *pluginsdk.Resource {
+	return &pluginsdk.Resource{
+		Create: resourceElasticTagRuleCreateorUpdate,
+		Read:   resourceElasticTagRuleRead,
+		Update: resourceElasticTagRuleCreateorUpdate,
+		Delete: resourceElasticTagRuleDelete,
+
+		Timeouts: &pluginsdk.ResourceTimeout{
+			Create: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Read:   pluginsdk.DefaultTimeout(5 * time.Minute),
+			Update: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Delete: pluginsdk.DefaultTimeout(30 * time.Minute),
+		},
+
+		Importer: pluginsdk.ImporterValidatingResourceId(func(id string) error {
+			_, err := parse.ElasticTagRuleID(id)
+			return err
+		}),
+
+		Schema: map[string]*pluginsdk.Schema{
+			"name": {
```

Comment:
Property `name` normally is the name of this tag rule resource. And the monitor name should be like a property called `monitor_name`

Reply:
updated